crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -291,7 +303,8 @@ def noderepocheck
   rescue Crowbar::Error::StartStepRunningError,
          Crowbar::Error::StartStepOrderError,
          Crowbar::Error::EndStepRunningError,
-         Crowbar::Error::SaveUpgradeStatusError => e
+         Crowbar::Error::SaveUpgradeStatusError,
+         StandardError => e
```

Comment:
Also, the point of catching StandardError is to end the step with failure, and this is not done when you are catching it here from controller. As a result, if there's a syntax error in the **model** code of nodesrepocheck, not only the error won't be reraised (so I just know `e.message` which is not enough), but also there's no code that would end the nodesrepocheck step and it is still marked as running